icechunk/src/repository.rs

```diff
@@ -139,8 +139,7 @@ pub struct Repository {
     asset_manager: Arc<AssetManager>,
     virtual_resolver: Arc<VirtualChunkResolver>,
     virtual_chunk_credentials: HashMap<ContainerName, Credentials>,
-    #[serde(skip)]
-    default_commit_metadata: Arc<RwLock<Option<SnapshotProperties>>>,
+    default_commit_metadata: Arc<Mutex<Option<SnapshotProperties>>>,
```

Comment:
Why do we need locking? Can't the `set_default_commit_metadata` just take a `&mut self`? I think that would be the cleanest option, then `PyRepository` is the one that needs locking to call this method on the repo, which seems also right.

Reply:
Because we dont currently lock `PyRepository` at all and its a big change to introduce it. We use internal mutability everywhere

Reply:
Sorry thats what i meant. I was trying to avoid having to lock the `Repository` but i can do if we want it

Reply:
We can merge and I can give it a try tomorrow if you are busy. But, i'd prefer if we don't introduce internal mutability into the Repo.

Reply:
i can do it in this pr then! Thanks for the feedback